### PR TITLE
[FFM-9959]: Implement featureflag tagging for terraform

### PR DESCRIPTION
**Pull request by ribeirophillipe**
## Describe your changes
Include tags as an extensions, so we add a new instruction for tags
Changes were also made to include the identifier and unit tests.
Include documentation and change log and examples

## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

File: internal/service/platform/feature_flag/resource_feature_flag.go [hunk 1]
```diff
@@ -98,6 +98,25 @@ func ResourceFeatureFlag() *schema.Resource {
...
+						"action": {
+							Description: "The action for the tag",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"name": {
+							Description: "The name of the tag",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
```
Comment (danmack-harness) [inline internal/service/platform/feature_flag/resource_feature_flag.go:116]: Tags also have an identifier too - does this need to captured here?

File: internal/service/platform/feature_flag/resource_feature_flag.go [hunk 2]
```diff
@@ -642,6 +665,28 @@ func buildFFPatchOpts(d *schema.ResourceData) *nextgen.FeatureFlagsApiPatchFeatu
...
+				if strings.ToLower(tagMap["action"].(string)) == "add" {
+					kind = aws.String(AddTag)
+				} else if strings.ToLower(tagMap["action"].(string)) == "remove" {
+					kind = aws.String(RemoveTag)
+				}
+				instruction := &Instruction{
+					Kind: kind,
+					Parameters: &Parameter{
+						Name:       aws.String(tagMap["name"].(string)),
+						Identifier: aws.String(tagMap["name"].(string)),
```
Comment (danmack-harness) [inline internal/service/platform/feature_flag/resource_feature_flag.go:682]: identifier is not always guaranteed to be the name - is name all we have?

**Review by danmack-harness — COMMENTED** [review]
I think the approach is correct but we need to capture the identifier of the tag too - is this possible?

File: internal/service/platform/feature_flag/resource_feature_flag.go (hunk 1 above)
Comment (ribeirophillipe) [reply internal/service/platform/feature_flag/resource_feature_flag.go:116]: I updated the schema to include the identifier

File: internal/service/platform/feature_flag/resource_feature_flag.go (hunk 2 above)
Comment (ribeirophillipe) [reply internal/service/platform/feature_flag/resource_feature_flag.go]: I've updated this, after including identifier in the schema.

**Review by danmack-harness — COMMENTED** [review]
I'm not sure that adding the action is the correct thing to do. I think that we need a DB read to retrieve all tags and either calculate deltas or remove all DB entries and then add TF entries. Have I misunderstood the Action part of the TF?

**Review by danmack-harness — APPROVED** [review]

**Pull request merged**